src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -686,38 +951,43 @@ function AnalyticsService:LogResourceEvent(data: ResourceEvent): { [any]: any }
 		elseif typeof(data.amount) ~= "number" then
 			return reject("amount is required")
 		end
-		
-		GameAnalytics:addResourceEvent(data.userId, {
+
+		if not table.find(self._resourceEventTypes, data.eventType) then
+			self._resourceEventTypes[#self._resourceEventTypes + 1] = data.eventType
+			GameAnalytics.Events:setAvailableResourceItemTypes(self._resourceEventTypes) -- Update the SDK
+		end
+
+		self:_wrapper("addResourceEvent", data.userId, {
 			-- FlowType is Sink by default
 			flowType = data.flowType,
 			currency = data.currency,
 			amount = data.amount,
 			itemType = data.eventType,
-			itemId = data.itemId
+			itemId = data.itemId,
 		})
-		
-		return resolve()
+
+		resolve()
 	end)
 end
 
 --[=[
 	Used to log errors
-	
+
 	Example Use:
 	```lua
 	local missionName: string = "Invalid Mission Name"
-	
+
 	AnalyticsService:LogError({
 		userId = player.UserId,
 		message = "Player tried to join a mission that doesn't exist named " .. missionName,
 		severity = "Error"
 	})
 	```
-	
+
 	@param data ErrorEvent
-	@return { [any]: any }
+	@return Promise
 ]=]
-function AnalyticsService:LogError(data: ErrorEvent): { [any]: any }
+function AnalyticsService:LogError(data: ErrorEvent)
```

Comment:
```suggestion
function AnalyticsService:LogError(data: ErrorEvent): {[any]: any}
```
Outside of a custom defined promise type, {[any]: any} is the closest to a return type for a promise in my opinion.